### ISSUE TITLE
.to_image() treats a stream as a regular file

### ISSUE THREAD

**Issue by Urbener** [jsvine/pdfplumber#948]
When working with a `ZipExtFile`, calling `page.to_image()` ends up throwing a `FileNotFoundError`, as it's treating the file name inside the zip file as a regular, filesystem-backed file. I think this will apply to other stream types as well, but I haven't been able to test it.

Calling `pypdfium2.PdfDocument._process_page()` directly works as expected, so I think the problem can be traced [here](https://github.com/jsvine/pdfplumber/blob/28f4ebeb3c0266cdc83fbfb73fb435fce55bcd15/pdfplumber/display.py#L44).

However, calling `pdfplumber.open(repair=True)` does work with _some_ files, and the types received by the `get_page_image()` change: with `repair=True`, it gets a `_io.BytesIO` object, while without it, it gets a `ZipExtFile`.
```
# reproducer.py
from zipfile import ZipFile

import pdfplumber

with ZipFile('reproducer.zip') as zip_file:
    with zip_file.open('dummy.pdf') as pdf_file:
        with pdfplumber.open(pdf_file) as pdf:
            page = pdf.pages[0]
            im = page.to_image()
```
Sample ZIP and PDF file:
[reproducer.zip](https://github.com/jsvine/pdfplumber/files/12181335/reproducer.zip)

Environment:
- pdfplumber version: 0.10.1
- Python version: 3.10.6
- OS: Linux (Ubuntu 22.04.2)

**Comment by jsvine** [comment]
Hi @Urbener, and thank you for flagging. Thanks, too, for the clear description, example file, and code to reproduce. Exactly the kind of issue I like to see!

I agree with your diagnosis of the issue / code to blame. I have some potential solutions in mind, which I'll test. Will keep you updated here.

**Comment by jsvine** [comment]
Thanks again! Should be fixed in https://github.com/jsvine/pdfplumber/commit/30a52cb and now available in v0.10.2

**Comment by Urbener** [comment]
It's my pleasure to help improve such a useful project. It should be me thanking you for creating it and your continued support.

Thanks!
